### ISSUE TITLE
SQLAlchemy Version impacts the behaviour of /api/cars/{car_id}/trips endpoint

### ISSUE THREAD

**Issue by SashiDareddy** [codesensei-courses/fastapi_fundamentals#2]
TLDR;
If you get the error `AttributeError: 'Car' object has no attribute 'trips'`, make sure you are using the correct SQLAlchemy version for your Python environment - don't use the latest available package (especially if you are not using Python 3.10).

It looks like currently the max version of SQLAlchemy that supports the `/api/cars/{car_id}/trips` endpoint as expected is `1.4.35`:

```shell
curl -X 'POST' \
  'http://localhost:8000/api/cars/2/trips' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "start": 0,
  "end": 100,
  "description": "string"
}
```

which returns:

```shell
{
  "id": 21,
  "start": 0,
  "end": 100,
  "description": "string",
  "car_id": 2
}
```

However if you use SQLAlchemy==1.4.36 or SQLAlchemy==1.4.37 (the latest available) then you get the following error when you issue the above mentioned `curl` command - I've observed this behaviour in Python `3.9.6` but in `3.10.4` you can use the latest version of SQLAlchemy 1.4.37:

```shell
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 366, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\fastapi\applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\middleware\cors.py", line 92, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\middleware\cors.py", line 147, in simple_response
    await self.app(scope, receive, send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\exceptions.py", line 93, in __call__
    raise exc
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\fastapi\middleware\asyncexitstack.py", line 21, in __call__
    raise e
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\fastapi\middleware\asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\routing.py", line 65, in app
    response = await func(request)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\fastapi\routing.py", line 227, in app
    raw_response = await run_endpoint_function(
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\fastapi\routing.py", line 162, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "C:\Users\sashi\AppData\Local\pypoetry\Cache\virtualenvs\fastapi-fundamentals-VQjl57SV-py3.9\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "D:\projects\Learning\PluralsightCourses\fastapi_fundamentals\.\routers\cars.py", line 80, in add_trip
    car.trips.append(new_trip)
AttributeError: 'Car' object has no attribute 'trips'
```

**Comment by SashiDareddy** [comment]
Fix for this issue is already suggested in the above post.
